packages/compose/src/hooks/use-resize-observer/index.native.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { View, StyleSheet } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Hook which allows to listen the resize event of any target element when it changes sizes.
+ *
+ * @return {Object} Measurements object with properties `width` and `height`.
+ *
+ * @example
+ *
+ * ```js
+ * const App = () => {
+ * 	const [ resizeListener, sizes ] = useResizeObserver();
+ *
+ * 	return (
+ * 		<View>
+ * 			{ resizeListener }
+ * 			Your content here
+ * 		</View>
+ * 	);
+ * };
+ * ```
+ *
+ */
+const useResizeObserver = () => {
+	const [ measurements, setMeasurements ] = useState( null );
+
+	const onLayout = useCallback( ( { nativeEvent } ) => {
+		const { width, height } = nativeEvent.layout;
+
+		setMeasurements( { width, height } );
```

Comment:
How often does `onLayout` get called? One thing about `setState` which is nice is that it avoids a render if the next value is the same. But in the case of an object like this, a new object will always be a new reference, so if `onLayout` is being called frequently, we may be updating more frequently than we want to be.

A few considerations:

- If it's only called in cases like a resize, then maybe it's fine as is.
- A simple change could be to manually compare current and next widths before calling `setMeasurements`
- Or, we could refactor this to two separate state values to leverage React's internal reconciliation

Reply:
Actually, it will be called frequently, so I should avoid it! I can do it in the same way as I had done before in the previous approach: will save in state current value and call `onLayout` only when the width is different than state value, so it will be following your idea:

> If it's only called in cases like a resize, then maybe it's fine as is.

Reply:
Actually there is no need for saving the current value in the state, only need to use a callback in `setState`